features/issues/components/issue-filters/issue-filters.tsx

```diff
@@ -0,0 +1,141 @@
+import { IssueLevel, IssueStatus } from "@features/issues/types/issue.types";
+import { Input, SelectComponent } from "@features/ui";
+import { space } from "@styles/theme";
+import { NextRouter, useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import { useDebounceValue } from "./use-debounce-value";
+
+interface OptionType {
+  label: string;
+  value: string;
+}
+
+// React-select deals in unknown option types. This is used to safely cast the select input values to our option type.
+function isOptionTypeValid(option: unknown): option is OptionType {
+  return option !== null && typeof option === "object" && "value" in option;
+}
+
+// Uses the Record type to allow us to index via string names
+const statusNames: Record<string, string> = {
+  open: "Unresolved",
+  resolved: "Resolved",
+};
+
+const levelNames: Record<string, string> = {
+  info: "Info",
+  warning: "Warning",
+  error: "Error",
+};
+
+const statusOptions: OptionType[] = [
+  { label: "Resolved", value: IssueStatus.resolved },
+  { label: "Unresolved", value: IssueStatus.open },
+];
+
+const levelOptions: OptionType[] = [
+  { label: "Info", value: IssueLevel.info },
+  { label: "Warning", value: IssueLevel.warning },
+  { label: "Error", value: IssueLevel.error },
+];
+
+const Container = styled.div`
+  display: grid;
+  grid-template-columns: 10rem 10rem 17.5rem;
+  align-items: center;
+  justify-content: end;
+  gap: ${space(4)};
+  padding-bottom: 1.125rem;
+`;
+
+// Defined outside the component to make useCallback/useEffect dependency decisions easier to understand
+const updateQueryParams = (
+  router: NextRouter,
+  param: string,
+  newValue: string | null
+) => {
+  if (newValue) {
+    router.push({
+      pathname: router.pathname,
+      query: { ...router.query, [param]: newValue },
+    });
+  } else {
+    delete router.query[param];
```

Comment:
This mutates the `router.query` object which I would discourage since it might lead to bugs (except if you saw it in the docs). As an alternative, you can create a new query object and delete the param from there. E.g.

```
const query = { ...router.query };
delete query[param];

router.push({
  pathname: router.pathname,
  query,
});
```

Reply:
This is a great point that I didn't consider. Adjusted.